File: tests/dotnet/TestNumber.cs
using Number_c;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace TestFFIG
{
    [TestClass]
    public class TestNumber
    {
        [TestMethod]
        public void NumberValue()
        {
          var number = new Number(8);
          
          Assert.AreEqual(number.value(), 8);
        }
        
        [TestMethod]
        public void NumberNext()
        {
          var number = new Number(8);
          var next_number = number.next();
          
          Assert.AreEqual(next_number.value(), 9);
        }
    }
}

File: tests/java/TestNumber.java
import static org.junit.Assert.assertEquals;
import org.junit.Test;
import Number.Number;

public class TestNumber {
    @Test
    public void Value()
    {
      Number number = new Number(8);

      assertEquals(number.value(), 8);
    }
    
    @Test
    public void NextNumberHasNextValue()
    {
      Number number = new Number(8);
      Number next = number.next();

      assertEquals(next.value(), 9);
    }
}
